smtlib-backends-z3/src/SMTLIB/Backends/Z3.hs

```diff
@@ -40,6 +40,20 @@ newtype Config = Config
   { -- | A list of options to set during the solver's initialization.
     -- Each pair is of the form @(paramId, paramValue)@, e.g.
     -- @(":produce-models", "true")@.
+    --
+    -- Note that Z3 has different kinds of parameters, and not all of
+    -- them can be set here. In particular, there are the so called
+    -- global and module parameters with a value that affects
+    -- all solver instances (in our experiments those started after
+    -- changing the parameter). We have found some of these global
+    -- parameters to be ignored when provided here. You might have more
+    -- luck setting them after starting the solver:
+    --
+    -- > command_ solver "(set-option :parameter_name value)"
+    --
+    -- Or using `Z3_global_param_set` from the Z3 API directly.
```

Comment:
This does suggest some kind of mitigation, eventually. Maybe there could be a way to use the appropriate `set` function for various options.

Reply:
Maybe. But the fact that global options affect all solver instances would deserve a separate call to make the scope more obvious.